File: dm-authority-common/src/main/java/com/dm/auth/controller/ScopeController.java
package com.dm.auth.controller;

import com.dm.auth.service.ResourceService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * scope
 */
@RequestMapping({"scopes", "p/scopes"})
@RestController
public class ScopeController {

    private final ResourceService resourceService;

    public ScopeController(ResourceService resourceService) {
        this.resourceService = resourceService;
    }

    /**
     * 获取所有的scope
     *
     * @return 获取到的scope的列表
     */
    @GetMapping
    public List<String> listScope() {
        return resourceService.listScopes();
    }
}

File: sample-servers/authorization-server/src/main/kotlin/com/dm/server/authorization/controller/ScopeController.kt
package com.dm.server.authorization.controller

import com.dm.server.authorization.service.ClientService
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController


@RestController
@RequestMapping("scopes")
class ScopeController(
    val clientService: ClientService
) {
    @GetMapping
    fun findScopes(
        @RequestParam(value = "keyword", required = false, defaultValue = "") keyword: String,
        pageable: Pageable
    ): Page<String> {
        return clientService.findScopes(keyword, pageable)
    }
}
